### ISSUE TITLE
[1.12.2] Cascading Worldgen

### ISSUE THREAD

**Issue by Shahelion** [Ordinastie/MalisisCore#141]
[16:50:54] [Server thread/WARN] [FML]: MalisisCore loaded a new chunk (97, 55  Dimension: 0) during chunk population, causing cascading worldgen lag. Please report this to the mod's issue tracker. This log can be disabled in the Forge config.

**Comment by labpaw** [comment]
@Ordinastie can you take a look at this? I think the issue is [here](https://github.com/Ordinastie/MalisisCore/blob/e6207770ac7248a6d25d27d30b3bf8b5956bb467/src/main/java/net/malisis/core/util/raytrace/RaytraceChunk.java#L65) (maybe somewhere else more)
Also check this: https://www.reddit.com/r/feedthebeast/comments/5x0twz/investigating_extreme_worldgen_lag/
It happens in Nether, just try to fly away for long distance.

**Comment by Ordinastie** [comment]
This is really a non-issue for MalisisCore. There is no world gen in MalisisCore, so there is no risk of uncontrolled cascading like shown in the thread.
MalisisCore might trigger the load (and generation) of an extra chunk but it won't cause lag problems.
Furthermore, Forge doesn't provide any context information because it assumes the source is world generation, so I can't even know from where it was triggered.

**Comment by Ricket** [comment]
Is it possible to fix it anyway? While exploring the overworld it really spams the server logs...

```
[00:40:28] [Server thread/WARN] [FML]: MalisisCore loaded a new chunk (104, 60  Dimension: 0) during chunk population, causing cascading worldgen lag. Please report this to the mod's issue tracker. This log can be disabled in the Forge config.
[00:40:31] [Server thread/WARN] [FML]: MalisisCore loaded a new chunk (87, 62  Dimension: 0) during chunk population, causing cascading worldgen lag. Please report this to the mod's issue tracker. This log can be disabled in the Forge config.
[00:40:31] [Server thread/WARN] [FML]: MalisisCore loaded a new chunk (89, 62  Dimension: 0) during chunk population, causing cascading worldgen lag. Please report this to the mod's issue tracker. This log can be disabled in the Forge config.
[00:40:31] [Server thread/WARN] [FML]: MalisisCore loaded a new chunk (90, 62  Dimension: 0) during chunk population, causing cascading worldgen lag. Please report this to the mod's issue tracker. This log can be disabled in the Forge config.
[00:40:32] [Server thread/WARN] [FML]: MalisisCore loaded a new chunk (103, 62  Dimension: 0) during chunk population, causing cascading worldgen lag. Please report this to the mod's issue tracker. This log can be disabled in the Forge config.
[00:40:34] [Server thread/WARN] [FML]: MalisisCore loaded a new chunk (95, 63  Dimension: 0) during chunk population, causing cascading worldgen lag. Please report this to the mod's issue tracker. This log can be disabled in the Forge config.
[00:40:35] [Server thread/WARN] [FML]: MalisisCore loaded a new chunk (98, 63  Dimension: 0) during chunk population, causing cascading worldgen lag. Please report this to the mod's issue tracker. This log can be disabled in the Forge config.
```
etc etc etc

FTB Revelation pack 1.3.0
malisiscore-1.12.2-6.3.1

**Comment by undeadbanana** [comment]
It's happening for me also, same version as Ricket.

**Comment by Ordinastie** [comment]
So I looked into it, and realised that Forge just takes a wild guess about which mod causes the cascading worldgen. I'm not responsible for the cascading, which makes sense since I don't do any worldgen at all.
You'll have to wait for Forge to have better reporting to really know what's going on.

**Comment by Ricket** [comment]
Ok, so we need to file a bug on Forge then, that their log message is wrong? Is there already such a bug or should I open one?

Can you link to anything demonstrating the "wild guess" algorithm that should be improved, so that I can include it in the bug report?

**Comment by labpaw** [comment]
@Ordinastie I think I found out why MalisisCore appears in logs. If any mod has cascading worldgen, some of his logs replaced with Malisis for some reason (In my case it was binnie mods and some other mod). If there are no other mods with cascading gen, there are no logs from MalisisCore. There also no logs if you running only Malisiscore and doors.

As for how to fix log spam at least, is it possible to prevent cascading logging for MalisisCore? BoP team done this for their mod (https://github.com/Glitchfiend/BiomesOPlenty/commit/765811f989aa02cb260a3f2ad808c192f1636198)
I guess it just says to forge to not log any possible cascading gen. Should be simple enough I guess.

**Comment by Ordinastie** [comment]
@Adaptivity Not quite. Forge tries to guess what mod causes it by going up the calling stack trace and finding the first first line that matches a loaded mod. Not really accurate enough.

What BoP does, is disable the logging while they do their world gen, and enable it back once they're done. Not really applicable here.
You can still disable that logging entirely yourself if you want, because it's available via Forge config (as stated in the log).

**Comment by undeadbanana** [comment]
Thanks for looking into this, I actually looked a little after I commented and can confirm what you said. What others said was that something is causing lag, and whatever mod made the last changes usually gets a finger pointed at them even if they don't do any world gen.
